src/pagination/pagination.tsx

```diff
@@ -156,6 +168,20 @@ const Component = (
     const onBlurLeftButton = () => {
         setHoverLeftButton(false);
     };
+
+    const handleListItemClick = (item: DropdownItemProps) => {
+        setSelecctedOption(item);
+        const pagesize = item.value;
+        setPageSize(pagesize);
+        totalPages = Math.ceil(totalItems / pagesize);
+        const page = activePage >= totalPages ? totalPages : activePage;
+
+        if (onPageSizeChange) {
+            onPageSizeChange(page, pagesize);
+        }
+        isFirstPage = activePage === 1;
+        isLastPage = activePage === totalPages;
```

Comment:
shouldn't be mutating local variables. the value is lost on the next render anyway. if you do need to persist it, use `useState` or `useRef`